apache-superset.md

```diff
@@ -1,86 +1,285 @@
-- git checkout -b mybranch $latest-upstream-tag.
-- Find list of commits that are Dimagi specific by doing `git log <superset-tag>..<dimagi-tag>`
-- Cherry-pick all the above commits onto mybranch
-- Create a new release for the new version by 'git tag  <apache-superset-release-tag>-dimagi-<index>'
+#### Example of adding a new customization
 
-If we take the example of 1.4.1-dimagi-1 for the latest released tag, the commands will be following
-- git fetch upstream (let's say the latest upstream is 1.5.1)
-- git tag -l "*dimagi*"
-  1.3.1-dimagi-1
-  1.3.1-dimagi-2
-  1.4.1-dimagi-1
-  1.4.1-dimagi-2
-- git checkout -b mybranch 1.4.1-dimagi-2
-- git log 1.4.1..1.4.1-dimagi-2 (shows all the commits containing our patches)
-- Do "git cherr-pick" for all of above commits.
-- git tag 1.5.1-dimagi-1
-- git push origin 1.5.1-dimagi-1
+Here is how to add a new customization to an existing Dimagi Superset
+release.
 
-Now, 1.4.1-dimagi-2 PyPi package can be built using the section below.
+1. Print all the tags containing "dimagi".
+   ```bash
+   $ git tag -l "*dimagi*"
+   1.3.1-dimagi-1
+   1.3.1-dimagi-2
+   1.4.1-dimagi-1
+   ```
+   Pick the latest one, "1.4.1-dimagi-1".
 
-### Building dimagi-superset Pypi package.
+2. If a branch at that tag doesn't already exist, then check out a new
+   branch (e.g. "my_custom") at that tagged commit.
+   ```bash
+   $ git checkout -b my_custom 1.4.1-dimagi-1
+   ```
 
-The Superset repo has a setup.py but it can not be directly pip installed as there is superset-frontend that needs to be built and shipped with the Python package. The exact set of instructions live at https://github.com/apache/superset/tree/master/RELEASING#publishing-a-convenience-release-to-pypi. At the point of writing this document, below are the instructions on how to build a Pypi package from this. You may need to make sure that the build instructions have remained the same before proceeding.
+3. Add new commits containing the new customizations, and open a pull
+   request with the tag "1.4.1-dimagi-1" as its base.
+
+4. Once merged, create a new tag with the index incremented, and push it
+   to GitHub.
+   ```bash
+   $ git tag 1.4.1-dimagi-2
+   $ git push origin 1.4.1-dimagi-2
+   ```
+
+Now a PyPI package can be built for "1.4.1-dimagi-2". See
+[instructions to do that](#building-dimagi-superset-pypi-package) later
+in this document.
+
+
+#### Example of applying translations
+
+Managing translations requires two tools:
+
+- **msgmerge**, which is probably already be installed. It is included
+  in the **gettext** package, which is one of the pre-install
+  requirements for CommCare HQ dev environments on Linux.
+
+- **po2json**, which needs to be installed using **npm**.
+
+You will also need Dimagi's Portuguese translations. You can reach out
+to members of the Solutions Division who work on projects in Mozambique
+for an updated version. The current version should be named
```

Comment:
If possible, good to take a note of where/how this translation is done, may be on confluence and link that here.
Assuming if someone needs to do this in future, they would be confused on where to look next after reading this.

Reply:
Great idea! I don't think that exists yet, but I'll reach out, and see if they can create that Confluence page.